testdata/main.go

```diff
@@ -6,32 +6,122 @@ import (
 	"time"
 	"context"
 	"strings"
+	"os"
+	"encoding/json"
 
 	a "github.com/ibmdb/go_ibm_db"
 )
 
 var ctx = context.Background()
 
-var host = "<HOST>"
-var port = "<PORT>"
-var database = "<DATABASE>"
-var uid = "<UID>"
-var pwd = "<PWD>"
+var host string
+var port string
+var database string
+var uid string
+var pwd string
+var connStr string
 
-var connStr = "PROTOCOL=tcpip;HOSTNAME=" + host + ";PORT=" + port + ";DATABASE=" + database + ";UID=" + uid + ";PWD=" + pwd
+//Read Config variable from json file
+type Config  struct {
+        Host string `json:"HOSTNAME"`
+        Port string `json:"PORT"`
+	Database string `json:"DATABASE"`
+	Uid string `json:"UID"`
+	Pwd string `json:"PWD"`
+}
+
+func LoadConfiguration(filename string) (Config, error) {
+        var config Config
+        configFile, err := os.Open(filename)
+        defer configFile.Close()
+        if err != nil {
+                return config, err
+        }
+        jsonParser := json.NewDecoder(configFile)
+        err = jsonParser.Decode(&config)
+        return config, err
+}
+
+//Get connection information from config.json
+func GetConnectionInfoFromConfigFile() {
+       config, _:= LoadConfiguration("config.json")
+       host = config.Host
```

Comment:
Only read from config file if value is not set by env var. Env var setting will have more priority. Env var might have set port but not host. So, update code accordingly. Thanks.